app/com/linkedin/drelephant/analysis/AnalyticJobGeneratorHadoop2.java

```diff
@@ -159,21 +197,80 @@ public void configure(Configuration configuration)
     logger.info("The failed apps URL is " + failedAppsURL);
     appList.addAll(failedApps);
 
-    // Append promises from the retry queue at the end of the list
-    while (!_retryQueue.isEmpty()) {
-      appList.add(_retryQueue.poll());
-    }
+    updateCheckPoint();
 
-    _lastTime = _currentTime;
     return appList;
   }
 
   @Override
-  public void addIntoRetries(AnalyticJob promise) {
-    _retryQueue.add(promise);
-    int retryQueueSize = _retryQueue.size();
-    MetricsController.setRetryQueueSize(retryQueueSize);
-    logger.info("Retry queue size is " + retryQueueSize);
+  public void analyseJob(AnalyticJob analyticJob) {
+    try {
+      String analysisName = String.format("%s %s", analyticJob.getAppType().getName(), analyticJob.getAppId());
+      long analysisStartTimeMillis = System.currentTimeMillis();
+      logger.info(String.format("Analyzing %s", analysisName));
+      AppResult result = analyticJob.getAnalysis();
+      result.save();
+      long processingTime = System.currentTimeMillis() - analysisStartTimeMillis;
+      logger.info(String.format("Analysis of %s took %sms", analysisName, processingTime));
+      MetricsController.triggerJobCompletedEvent(analyticJob.getRetriesCount(), processingTime);
+
+    } catch (InterruptedException e) {
+      logger.info("Thread interrupted");
+      logger.info(e.getMessage());
```

Comment:
We need to handle all exceptions. Either they should go to retry or it should get failed.